src/ui/IconButton.js

```diff
@@ -0,0 +1,74 @@
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import FontAwesome from 'react-fontawesome';
+import {
+  Button,
+  OverlayTrigger,
+} from 'react-bootstrap';
+import * as Utils from '../utils';
+
+/**
+ * Component that renders a button with a Font Awesome Icon.
+ */
+const IconButton = ({
+  icon,
+  label,
+  tooltip,
+  placement,
+  className,
+  ...props
+}) => {
+  let button = (
+    <Button className={classNames('btn-icon', className)} {...props}>
+      <FontAwesome name={icon} />
+      {_.isEmpty(label) ? null : (
+        <span className="btn-icon-label">{label}</span>
+      )}
+    </Button>
+  );
+
+  if (tooltip) {
```

Comment:
I think we should avoid hardcoding wrapper components by the means of if logic, given that mixes components' responsibilities. In theory we'd need to do that on every button to keep a consistent API, so it makes more sense to make decorator/HOC out of the overlay. So that we don't have to maintain wrapper code on every custom button. Also, that decorator/HOC could be used in separate, which is definitely desirable.

Reply:
I agree with you on that one. Not sure if we should have a HOC or a plain React Component for the Tooltip. With HOC we would need to declare our button components like so:

```
import { tooltip, IconButton } from 'bandit-pouch';

const TooltipButton = tooltip('Tooltip text', 'top')(IconButton)

export default () => (
  <TooltipButton icon="plus" /> 
)
```
vs plain component:
import { Tooltip, IconButton } from 'bandit-pouch';

export default () => (
  <Tooltip text="Tooltip" placement="top">
    <IconButton icon="plus" /> 
  </Tooltip>
)
```

WYT?